### DISCUSSION TITLE
Welcome to xDiT Discussions!

### DISCUSSION THREAD

**Discussion by feifeibear** [xdit-project/xDiT#156] [Announcements]
Please let us know which DiT model you are most interested in.

**Comment by lsyyysky** [comment]
Now I am learning ai system and cuda.Please how could i join in this project?

**Reply by feifeibear** [reply]
Here are some suggested tasks to help you get started:

1. Implement some unit tests for the project and try to understand the attention mechanism implementations of different models.
Link: [https://github.com/xdit-project/xDiT/blob/main/xfuser/model_executor/layers/attention_processor.py](https://github.com/xdit-project/xDiT/blob/main/xfuser/model_executor/layers/attention_processor.py)

2. Support some new models and learn about their usage.

3. Optimize the code using methods like `torch.compile`, `onediff`, etc.

**Reply by lsyyysky** [reply]
Thanks!

**Comment by lsyyysky** [comment]
(base) ➜  xDiT git:(main) python ./examples/pixartalpha_example.py -h
/environment/miniconda3/lib/python3.10/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension: '/environment/miniconda3/lib/python3.10/site-packages/torchvision/image.so: undefined symbol: _ZN3c1017RegisterOperatorsD1Ev'If you don't plan on using image functionality from `torchvision.io`, you can ignore this warning. Otherwise, there might be something wrong with your environment. Did you have `libjpeg` or `libpng` installed before building `torchvision` from source?
  warn(
2024-08-21 11:31:07.491529: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-08-21 11:31:08.107901: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
Traceback (most recent call last):
  File "/home/featurize/work/xDiT/./examples/pixartalpha_example.py", line 7, in <module>
    from xfuser.core.distributed import (
ModuleNotFoundError: No module named 'xfuser.core'
(base) ➜  xDiT git:(main) python ./examples/pixartalpha_example.py -h
(base) ➜  xDiT git:(main) pip show xfuser                            
Name: xfuser
Version: 0.2
Summary: xDiT: A Scalable Inference Engine for Diffusion Transformers (DiTs) on multi-GPU Clusters
Home-page: https://github.com/xdit-project/xDiT.
Author: Jiannan Wang, Jiarui Fang, Jinzhe Pan, Aoyu Li, Pengcheng Yang
Author-email: fangjiarui123@gmail.com
License: 
Location: /environment/miniconda3/lib/python3.10/site-packages
Requires: accelerate, beautifulsoup4, diffusers, distvae, ftfy, sentencepiece, torch, transformers
Required-by: 
(base) ➜  xDiT git:(main) pip show tensorrt                          
Name: tensorrt
Version: 10.3.0
Summary: TensorRT Metapackage
Home-page: https://developer.nvidia.com/tensorrt
Author: NVIDIA Corporation
Author-email: 
License: Proprietary
Location: /environment/miniconda3/lib/python3.10/site-packages
Requires: tensorrt-cu12
Required-by:

**Reply by lsyyysky** [reply]
Please i have installed tensort and xfuser, but excueting " python ./examples/pixartalpha_example.py -h"  has not find tensort and xfuser.core. What do i need to do?

**Reply by feifeibear** [reply]
You should pip install from the latest version.
I guess you install it from pip.
I will update the pip version ASAP.

**Reply by lsyyysky** [reply]
thanks! i try again

**Reply by feifeibear** [reply]
I have upgrade the xDiT version to 0.3 and upload new wheel to pypi. You can install
pip install xfuser==0.3

**Comment by dukeqiao** [comment]
really need support for diffusers flux with controlnet

**Reply by feifeibear** [reply]
Thanks for your suggestion. Most of people use ControlNet+Flux.1 in ComfyUI. 

We just wonder how people use controllers with diffusers' flux. We are planing to implement controlnet in ComfyUI.

**Reply by dukeqiao** [reply]
the reason I trying to use diffusers fluxpipeline is the inference speed,   ComfyUI pipeline is too slow for online deployment

**Reply by feifeibear** [reply]
Would you like to help us add controlnet in fluxpipeline. We can draft a issue to give you some detail instructions?

**Reply by dukeqiao** [reply]
sure, in fact I already tried, but there is some problems with controlnet input shape with running   _sync_pipeline function

**Reply by Lay2000** [reply]
> sure, in fact I already tried, but there is some problems with controlnet input shape with running _sync_pipeline function

Hi, Duke! Could you please provide more details on the problem you met? Like the entry scripts with hyperparameters you used and the Error information.

**Comment by satcon25** [comment]
I think you might want to try publishing a step by step guide for home users? This might help the project gain more attention from those of us that utilize multi-gpu systems. For example I currently have a couple new 5070 TI a 3090 and a 4060 for a combined total of 76GB of VRAM. I would love to try your project using flux but would need step by step guidance on this. 

+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 570.124.04             Driver Version: 572.61         CUDA Version: 12.8     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 5070 Ti     On  |   00000000:01:00.0 Off |                  N/A |
|  0%   42C    P8             17W /  300W |     436MiB /  16303MiB |      2%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
|   1  NVIDIA GeForce RTX 3090        On  |   00000000:03:00.0 Off |                  N/A |
|  0%   35C    P8             12W /  350W |       0MiB /  24576MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
|   2  NVIDIA GeForce RTX 5070 Ti     On  |   00000000:17:00.0 Off |                  N/A |
|  0%   50C    P8             18W /  300W |      20MiB /  16303MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
|   3  NVIDIA GeForce RTX 4060 Ti     On  |   00000000:49:00.0 Off |                  N/A |
|  0%   28C    P8              2W /  165W |     624MiB /  16380MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

**Comment by eppaneamd** [comment]
Hello @feifeibear! Quick question: if one wants to run the `hunyuan_video_usp_example.py` with bs > 1, should this be reflected also in [the video input parameters](https://github.com/xdit-project/xDiT/blob/main/examples/hunyuan_video_usp_example.py#L237)?

**Reply by feifeibear** [reply]
because DiT inference is compute bound, we have never tested bs>1. We suggest you inference each a batch of inputs one by one.

**Comment by lavinal712** [comment]
Can you provide support for Magi-1?

https://github.com/SandAI-org/MAGI-1

**Comment by Emilyhelin** [comment]
Hello, I am very interested in your research. I am studying the parallelization of diffusion model for offline time series denoising. I would like to ask if this framework can be applied to my topic? I noticed that you are mainly serving real-time

**Reply by feifeibear** [reply]
The DiT serving is computing bound. So batch size=1 is the same efficiency as batch size > 1. So online and offline should apply the same solution.

**Reply by Emilyhelin** [reply]
That's gr8t :)
Will you continue to maintain this framework over the next year?

**Reply by lavinal712** [reply]
> The DiT serving is computing bound. So batch size=1 is the same efficiency as batch size > 1. So online and offline should apply the same solution.

I also wonder that.

**Reply by eppaneamd** [reply]
> That's gr8t :) Will you continue to maintain this framework over the next year?

Has there been some developments regarding how actively this framework is currently maintained and/or will be maintained in the future? 🙏

**Comment by lavinal712** [comment]
Hello, recently I have been looking at the CFG parallel methods and trying to apply xdit to lumina-nextdit. I found that there are some discrepancies between the CFG parallel in the tutorial and the actual code. Could you provide some guidance on how to proceed?

**Reply by feifeibear** [reply]
could you provide more details about how to reproducing your problem?

**Reply by lavinal712** [reply]
We can discuss in https://github.com/xdit-project/xDiT/issues/519

**Comment by caizhi-mt** [comment]
方便分享一下xDiT下一个版本roadmap吗，或者开发计划？ @feifeibear

**Comment by yishuihanbaobao** [comment]
有微信交流群么？这里讨论不太及时呢。
